lib/shopify-cli/shopifolk.rb

```diff
@@ -0,0 +1,75 @@
+module ShopifyCli
+  ##
+  # ShopifyCli::Shopifolk contains the logic to determine if the user appears to be a Shopify staff
+  #
+  # The Shopifolk Feature flag will persist between runs so if the flag is enabled or disabled,
+  # it will still be in that same state until the next class invocation.
+  class Shopifolk
+    GCLOUD_CONFIG_PATH = '~/.config/gcloud/configurations/config_default'
+    DEV_PATH = '/opt/dev'
+    SECTION = 'core'
+    FEATURE_NAME = 'shopifolk'
+    def self.check
+      ##
+      # will return if the user appears to be a Shopify employee, based on several heuristics
+      #
+      # #### Returns
+      #
+      # * `is_shopifolk` - returns true if the user is a Shopify Employee
+      #
+      # #### Example
+      #
+      #     ShopifyCli::Shopifolk.check
+      #
+      ShopifyCli::Shopifolk.new.shopifolk?
+    end
+
+    def shopifolk?(gcloud_config_path = GCLOUD_CONFIG_PATH, dev_path = DEV_PATH)
+      ##
```

Comment:
Since we're persisting to the config, we can return early from this rather than re-checking the config, e.g. 

`return true if Feature.enabled?(FEATURE_NAME)`

Reply:
good call!

Reply:
Oh, are we concerned that by doing this the feature will never be disabled?

Reply:
That is true, but I can't think of a scenario where we'd need to.

(and if we didn't the config file could still be manually edited).

Reply:
@andyw8 I added in your suggestion, but I also added an extra argument to optionally skip this. Mostly because I think it makes sense to test this file without the 'always return true' logic. Let me know if that's appropriate!

Reply:
I'd say stubbing would be preferable here, e.g.:

`Features.stubs(:enabled?).with('shopifolk').returns...`